packages/tools/kadena-cli/README.md

```diff
@@ -716,27 +736,121 @@ Tool for creating and managing transactions
-| --out-file              | Provide path to save output                    |              |
-| --xx                    | Args are created onl the fly based on template |              |
-
-example:
-
-```
-kadena tx add --template="transfer.yaml" --template-data="path" (--account-from="k:account" --account-to="k:toaccount" --amount="1" --chain-id="1" --pk-from="publicKey") --network-id="testnet04" --out-file="transaction-(request-key).json"
-```
-
-Template variables are prompted unless assigned via --template-data or direct
-flags. Variables can be prefixed to enable value selection using:
-
-- `account:` to select a account name
-- `key:` to select a public key
-- `network:` to select existing networkId
-- `decimal:` to enable decimal input validation
+`kadena tx add` is a powerful command that leverages transaction templates to
+facilitate the quick and efficient creation of transactions across multiple
+chains and access patterns. This feature is designed to work with user-supplied
+values, filling out predefined templates to generate transactions ready for
+signing and submission.
+
+### Available Templates
+
+Currently, two default templates are provided: `transfer` and `safe-transfer`.
+These templates cover the most common transaction types, allowing for
+straightforward transfers of tokens between accounts.
+
+### Command Usage
+
+```plaintext
+kadena tx add [options] [arguments]
+```
+
+This command accepts various arguments and options, allowing for detailed
+customization of the transaction being created. Below is a breakdown of the
+options available:
+
+| **Arguments & Options** | **Description**                                           | Required |
+| ----------------------- | --------------------------------------------------------- | -------- |
+| `--template`            | Path to the transaction template file.                    | Yes      |
+| `--template-data`       | File path for the data used to fill the template.         | No       |
+| `--network-id`          | Specifies the network ID (e.g., `testnet04`).             | Yes      |
+| `--out-file`            | Path for saving the generated transaction file.           | No       |
+| `--holes`               | Displays a list of required template variables.           | No       |
+| Custom options          | Generated based on the chosen template's required fields. | Varies   |
+
+### Example Command
+
+```
+kadena tx add --template="transfer.yaml" --template-data="data.yaml" (--account-from="k:account" --account-to="k:toaccount" --amount="1" --chain-id="1" --pk-from="publicKey") --network-id="testnet04" --out-file="transaction.json"
+```
+
+In this example, `transfer.yaml` is the template used to construct the
+transaction. `data.yaml` contains the user-supplied values for the template
+variables. The `--network-id` specifies which network the transaction is
+intended for, and `--out-file` determines where the generated transaction file
+will be saved.
+
```

Comment:
before or after this, we can show an example of how transfer.yaml file and data.yaml looks like